stdlib/test/collections/test_list.mojo

```diff
@@ -84,6 +84,34 @@ def test_list():
     assert_equal(0, len(list))
     assert_equal(2, list.capacity)
 
+    # Test pop with index
```

Comment:
**Suggestion** Do you mind moving this into a separate function for testing this `pop` API functionality?

Reply:
will do